Part1_InterruptControllersEvolution.html

```diff
@@ -34,30 +34,30 @@ <h3>Introduction</h3>
 
 	<p>This article is about hardware/external interrupts IRQ.<p/>
 
-	<p>What is the purpose of interrupts? For example, we want to make some action with an incoming packet from the network card, as soon as the packet arrives. If you don't want to continuously ask the network card "Have my packet arrived?" and waste your processor time, you can make use of external hardware interrupt IRQ. Interrupt line from a device should be connected to INTR line of CPU, and after each packet receive, the network card will make a signal over this line. CPU will sense this signal, and know that network card has information for it. Only after that CPU will read the incoming packet.<p/>
+	<p>What is the purpose of interrupts? For example, we want to perform an action with an incoming packet from the network card as soon as the packet arrives. If you don't want to continuously ask the network card "Has my packet arrived?" and waste your processor time, you can use external hardware interrupt IRQ. The interrupt line from a device should be connected to the INTR line of the CPU, and after each packet is received, the network card will make a signal over this line. The CPU will sense this signal and know that the network card has information for it. Only after that will the CPU read the incoming packet.<p/>
 
-	<p>But what should we do if there are a lot of external devices? Cause there are can't be enough pins on CPU for all of them.<p/>
+	<p>But what should we do if there are more external devices than pins on the CPU? There won't be enough room for all of them.<p/>

```

Comment:
I don't think this rephrase is correct: "if there are more external devices than pins on the CPU?". CPU can have total of 500 pins, but 2 devices can't be handled with one INTR pin. And if CPU will have 2 INTR pins, CPU couldn't handle 3 devices (but total of CPU pins will still be 500). If this sentence needs rephrasing can it be done in another way?

Reply:
Oh yeah, that makes sense. Maybe then "But what should we do if there are not enough pins on the CPU for all the external devices?" is more accurate. In that case we could remove the next sentence, since it's clear then there's not enough room.

Reply:
Can it be like so?

```suggestion
	<p>But what should we do if there are a lot of external devices? Since there are can't be enough pins on the CPU for all of them.<p/>
```

Reply:
I would say "There won't be enough ...". "since" isn't used very often to start sentences

Reply:
Hmm... How about this rephrase?

```suggestion
	<p>But what should we do if there are a lot of external devices? It would be very unproductive to make a ton of INTR pins on the CPU for all of them.<p/>
```